docs/docs/classic-ml/tracking/server-security.mdx

```diff
@@ -0,0 +1,346 @@
+- **Cross-Origin Resource Sharing (CORS) Attacks**: Blocks unauthorized cross-origin requests that could steal data
+- **Host Header Injection**: Prevents attacks that manipulate the Host header for cache poisoning or password reset attacks
+- **Clickjacking**: Prevents embedding MLflow UI in malicious iframes
+- **MIME Type Confusion**: Blocks browsers from incorrectly interpreting file types
+
+## Default Security Behavior
+
+By default, MLflow server is configured for **secure local and internal network development**:
+
+- Accepts connections from localhost addresses (127.0.0.1, ::1, localhost) and private IP ranges (192.168._, 10._, 172.16-172.31.\*)
+- Blocks cross-origin requests except from common localhost ports (3000, 5000, 8000, 8080)
+- Validates Host headers to prevent DNS rebinding attacks
+- Adds security headers to all responses (X-Frame-Options, X-Content-Type-Options)
+
+This means you can run `mlflow server` on your local machine or internal network without any configuration, and team members can access it from private IP addresses.
+
+## Configuration Options
+
+Security is configured via environment variables:
+
+### Core Security Variables
+
+| Variable                        | Description                                  | Default                                  | Example                                                   |
+| ------------------------------- | -------------------------------------------- | ---------------------------------------- | --------------------------------------------------------- |
+| `MLFLOW_CORS_ALLOWED_ORIGINS`   | Comma-separated list of allowed CORS origins | localhost ports (3000, 5000, 8000, 8080) | `"https://app.company.com,https://dashboard.company.com"` |
+| `MLFLOW_ALLOWED_HOSTS`          | Comma-separated list of allowed Host headers | localhost + private IP ranges            | `"mlflow.company.com,mlflow.internal:5000"`               |
+| `MLFLOW_HOST_HEADER_VALIDATION` | Enable Host header validation                | `"true"`                                 | `"false"` (not recommended)                               |
+| `MLFLOW_ALLOW_INSECURE_CORS`    | Allow ALL origins (DANGEROUS!)               | `"false"`                                | `"true"` (dev only!)                                      |
+
+## Common Deployment Scenarios
+
+### 1. Local Development (Default)
+
+No configuration needed - just run:
+
+```bash
+mlflow server
+```
+
+The server will:
+
+- Accept connections only from localhost
+- Allow CORS from localhost origins (ports 3000, 5000, etc.)
+- Be protected against external attacks
+
+### 2. Team Development Server
+
+For a shared development server on your internal network:
+
+```bash
+# Allow team members to access from their browsers
+export MLFLOW_CORS_ALLOWED_ORIGINS="http://dev-server:5000,http://192.168.1.100:5000"
+export MLFLOW_ALLOWED_HOSTS="dev-server:5000,192.168.1.100:5000"
+
+mlflow server --host 0.0.0.0
```

Comment:
Setting both `MLFLOW_ALLOWED_HOSTS` and mlflow server `--host` option looks a bit confusing.


What about we remove mlflow server `--host` option,
instead,
we provide options like `--MLFLOW_ALLOWED_HOSTS` / `--MLFLOW_ALLOW_INSECURE_CORS` / `--MLFLOW_CORS_ALLOWED_ORIGINS`

Reply:
Yep this is super confusing. I love the idea of the cli args (added) and put better help messages to explain the purposes of these (host is different from allowed hosts)